src/components/UploadAvatar.jsx

```diff
@@ -18,7 +16,8 @@ const UploadAvatar = ({ label, name, file = '', setFieldValue, setSubmitting, is
     if (file) {
       const { url = '' } = await dashx.upload({
         file,
-        externalColumnId: externalColumnMapping['users.avatar']
+        resource: 'user',
+        attribute: 'avatar'
```

Comment:
When using `await` it's best practice to wrap it in `try...catch`. Alternatively you could use
```js
const handleChange = (event) => {
    const file = event.target.files[0]
    if (!file) return

    setSubmitting(true)
    dashx.upload({
      file,
      resource: 'user',
      attribute: 'avatar'
    }).then(({ url }) => {
      setFieldValue(name, url)
    }).catch(console.error)
    .finally(() => setSubmitting(false))
  }
```

Also noticing that `file` variable is being reused within same context. Can we change this component's signature to the following:

```js
const AvatarInput = ({ label, name }) => {
   const { setSubmitting, isSubmitting } = useFormikContext()
   const [ field, meta, helpers ] = useField(name);
   const { value } = meta; // Use this instead of the `file` prop
   const { setValue } = helpers; // Use this instead of the `setFieldValue` prop

   // rest of the code
}
```